src/librustc/ty/inhabitedness/mod.rs

```diff
@@ -113,9 +113,16 @@ impl<'a, 'gcx, 'tcx> AdtDef {
         tcx: TyCtxt<'a, 'gcx, 'tcx>,
         substs: SubstsRef<'tcx>) -> DefIdForest
     {
-        DefIdForest::intersection(tcx, self.variants.iter().map(|v| {
-            v.uninhabited_from(tcx, substs, self.adt_kind())
-        }))
+        let is_declared_nonexhaustive =
+            self.is_variant_list_non_exhaustive() && !self.did.is_local();
+        // Non-exhaustive enums from other crates are always considered inhabited.
+        if self.is_enum() && is_declared_nonexhaustive {
```

Comment:
`self.is_enum()` is redundant.

Reply:
Nit: the naming is a bit misleading, due to `is_local` `is_declared_nonexhaustive` can be false even if the enum was indeed declared non-exhaustive.